### PR TITLE
Allows the chaplains' null rod to be reskinned and renamed

### PR DESCRIPTION
**Pull request by ghost**
Adds the framework for weapons to be renamed and reskinned.

axe
fish
sword
sbf
sbb
sbf2
srh
ab
ab2
arh
fb

fb2
frh

### CODE REVIEW

**Comment by ghost** [comment]
Oh, and I'm dumb as a sack of bricks so I couldn't figure out how to add the sprites the easy way.

**Comment by SplergeBucket** [comment]
wot do the skins look rike

**Comment by ArcLumin** [comment]
You made your "wopit" give names for the "wopit" when reskinning. Make it use [name] rather than the word wopit

**Comment by ghost** [comment]
References to "wopit"s have been removed, and replaced with [name].

A couple of belt sprites were also backwards. Had to actually look pretty close to notice it, but they've been fixed.

**Comment by ArcLumin** [comment]
Do you have an image of the reskins? Put them in the opening paragraph

**Comment by ghost** [comment]
Images of reskins added

**Comment by Hackole** [comment]
The sprites look good and the idea is neat, +1/3

**Comment by ghost** [comment]
One thing I did was change the weapon weight from 1 to 3, because sticking a sword, pole axe, or great big fish in your pocket seemed a little odd. I'm wondering if it would be acceptable to give a minor damage buff as well, to compensate.

**Comment by danhibiki** [comment]
I don't think the null rod remotely needs a buff, but wouldn't setting it to weight to also stop you from pocketing it, but allow you to put it in backpacks?

**Comment by ghost** [comment]
Weight class three allows the item to fit into bags, but not boxes. I can change it to weight class two if you want it to fit into boxes though.

**Comment by ArcLumin** [comment]
I'd love to see the different skins get different properties

**Comment by ghost** [comment]
I tried, but it didn't work. I really wanted to give the bladed skins their own hitsounds. The method I used was just creating a new child item for the null rod for each of the options.

If anyone has another idea that might work with the way this is set up, I can give it a go.

**Comment by danhibiki** [comment]
If each one is a child, it would be as simple as assigning the new values to the children, right? In general, I don't think "reskins" should entail buffs though, as the idea of a reskin is to keep the original functionality, but change the look. Its pretty much semantics i'm talking about though

**Comment by ghost** [comment]
Yeah, I wanted to keep this visual only, and sound if possible. Really, my only concern at this point is getting the PR processed, and I don't much care what's done or not done to it.

**Comment by danhibiki** [comment]
Sound shouldn't be hard, that should be just defining hitsound for each child.

File: code/game/objects/weapons.dm [hunk 1]
```diff
@@ -2,6 +2,32 @@
...
+
+/obj/item/weapon/proc/reskin_wopit(mob/M)
+	var/choice = input(M,"Warning, you can only reskin your [name] once!","Reskin [name]") in moptions
+
+	if(src && choice && !M.stat && in_range(M,src) && !M.restrained() && M.canmove)
+		if(moptions[choice] == null)
+			return
+		else
+			icon_state = moptions[choice]
+			item_state = moptions[choice]
```
Comment (danhibiki) [inline code/game/objects/weapons.dm:26]: Around here is where you'd want to reset your weapon sounds. I'd just organize the soundbits to be named after their moptions name, and do something like "hitsound = '/sound/weapons/[moptions[choice]].ogg'" Not sure if you can use that syntax specifically for that function, but I'm fairly certain its possible. Or doing a length if statement, but that isn't wanted I assume.

**Comment by ghost** [comment]
I've got something I'll try out

**Comment by ghost** [comment]
Well, I can set up an if statement in weapons.dm if the hitsound thing is important, but it'll be ugly.

Unless something can be found to make Dan's idea work, and we're both stumped at the moment, I'd rather let it be.

**Comment by ghost** [comment]
Changed the left inhands to match the right ones. I always forget to do that.

I've also given up completely on giving the skins unique hitsounds. It would require making a very large mess, which is the exact opposite of what I'm trying to do here.

**Comment by LuisReyesX** [comment]
I would like the possessed sword from TG to be ported.

**Comment by doobiedoo23** [comment]
I'm pretty sure /tg/ did this already, might be worth checking out their code to see if their solution was similar to yours?

Anyway, +1/3

**Comment by ghost** [comment]
/tg/ has the code to rename weapons in weaponry.dm, versus weapons.dm where I have it.

Nothing under weapons for reskinning though.

**Comment by Kokojos** [comment]
Pretty nice +1

**Comment by Kokojos** [comment]
Testes live yesterday

**Pull request merged**